diff --git a/settingswindow.cpp b/settingswindow.cpp
index 7a0583f..1e1e366 100644
--- a/settingswindow.cpp
+++ b/settingswindow.cpp
@@ -759,6 +759,7 @@ void SettingsWindow::sendSignals()
     emit afterPlaybackDefault(Helpers::AfterPlayback(WIDGET_LOOKUP(ui->afterPlaybackDefault).toInt()));
 
     emit option("ytdl-format", QString("bestvideo[height<=%1]+bestaudio/best").arg(WIDGET_TO_TEXT(ui->ytdlpMaxHeight)));
+    emit option("ytdl-raw-options", "js-runtimes=quickjs,remote-components=ejs:github");
 
     emit zoomCenter(WIDGET_LOOKUP(ui->playbackAutoCenterWindow).toBool());
     double factor = WIDGET_LOOKUP(ui->playbackAutoFitFactor).toInt() / 100.0;
